core/api/full_node/make.cpp

```diff
@@ -335,6 +335,7 @@ namespace fc::api {
             auto &entropy) -> outcome::result<Randomness> {
       std::unique_lock ts_lock{*env_context.ts_branches_mutex};
       OUTCOME_TRY(ts_branch, TsBranch::make(ts_load, tipset_key, ts_main));
+      ts_lock.unlock();
```

Comment:
Why if it should be free after next line?

Reply:
because function called on next line (before destructor) needs `shared_lock` on this mutex.